pkg/logicalbackup/logicalbackup.go

```diff
@@ -692,7 +595,7 @@ func (b *LogicalBackup) prepareTablesForPublication(conn *pgx.Conn) error {
 
 		for rows.Next() {
 			tab := tableInfo{}
-			err = rows.Scan(&tab.oid, &tab.name.Namespace, &tab.name.Name, &tab.hasPK, &tab.replicaIdentity)
+			err := rows.Scan(&tab.oid, &tab.name.Namespace, &tab.name.Name, &tab.hasPK, &tab.replicaIdentity)
```

Comment:
not sure if  idiomatic go consider this acceptable, but I'd write
```
if rows.Scan(... ) != nil {
    return
```
as we don't need to preserve err.

Reply:
the problem here is that you're shadowing the `err` which is defined prior that line

Reply:
ah. now I've figured out that shadowing is intended here